client/components/newList-Form.jsx

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+
+export default class NewListForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      listName: ''
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleCancel = this.handleCancel.bind(this);
+  }
+
+  handleChange(event) {
+    const { name, value } = event.target;
+    this.setState({ [name]: value });
+  }
+
+  handleCancel(event) {
+    event.preventDefault();
+    window.location.hash = 'My-Lists';
+    this.setState({ listName: '' });
```

Comment:
You can skip setting state as React will destroy this component instance anyway and recreate it from scratch when the user returns to the form